datafusion 48

onursatici opened the pull request.

onursatici committed:
datafusion 48

Signed-off-by: Onur Satici <onur@spiraldb.com>

AdamGS commented on vortex-datafusion/src/lib.rs:
```diff
@@ -47,26 +49,24 @@ fn supported_data_types(dt: DataType) -> bool {
-            log::debug!("DataFusion expression can't be pushed down: {expr:?}");
-            false
-        }
+fn can_be_pushed_down(expr: &PhysicalExprRef, schema: &Schema) -> bool {
```
this is a nice

AdamGS commented on vortex-datafusion/src/lib.rs:
```diff
@@ -47,26 +49,24 @@ fn supported_data_types(dt: DataType) -> bool {
-            log::debug!("DataFusion expression can't be pushed down: {expr:?}");
-            false
-        }
+fn can_be_pushed_down(expr: &PhysicalExprRef, schema: &Schema) -> bool {
```
was just thinking last night due we support new things here? We have `In` now too right? (@joseph-isaacs?)

joseph-isaacs replied:
But its a pretty slow rn, what type would it run over?

AdamGS replied:
whatever is available I guess? if its not a perf win right now we can add that later

AdamGS commented:
I think the tests fails because datafusion used to default to `Utf8` and now it defaults to `Utf8View`, and that was implicit somewhere in our conversion logic

AdamGS commented:
IDK what happened to query 6, but we're 20x worse and parquet is 25x worse. There were a bunch of changes to statistics and repartitioning recently and I wonder if that's the issue there, the query is:
```sql
SELECT MIN("EventDate"), MAX("EventDate") FROM hits;
```

onursatici committed:
test

Signed-off-by: Onur Satici <onur@spiraldb.com>

onursatici committed:
bump deps

Signed-off-by: Onur Satici <onur@spiraldb.com>

robert3005 commented:
Wonder if they no longer correctly prune on file level statistics when they added partition group statistics

AdamGS commented:
@robert3005 thats my suspicion, parts of that were already in v47 but v48 has at least one more big change around that

onursatici commented:
I am also curious, will have a look, query 0 is also 10x @ parquet and 3x @ vortex

robert3005 commented:
I filed an issue upstream https://github.com/apache/datafusion/issues/16444

AdamGS commented:
I think this fine to merge, we can re-enable the datafusion config (+ #3569) and once apache/datafusion#16447 is released it'll be the default behavior again.

AdamGS committed:
Make sure the session config gets applied to our tables (#3569)

This is required for https://github.com/apache/datafusion/pull/16447 to
take effect (once its merged). By default datafusion will take care of
that if you use the SQL interface (and do all the session state setup).

Signed-off-by: Adam Gutglick <adam@spiraldb.com>

onursatici committed:
lock

Signed-off-by: Onur Satici <onur@spiraldb.com>
